yt_dlp/extractor/brilliantpala.py

```diff
@@ -0,0 +1,139 @@
+import hashlib
+
+from .common import InfoExtractor
+from ..utils import (
+    ExtractorError,
+    dict_get,
+    traverse_obj,
+    urlencode_postdata,
+)
+
+
+class BrilliantpalaIE(InfoExtractor):
+    _NETRC_MACHINE = 'brilliantpala'
+    _DOMAIN = '{subdomain}.brilliantpala.org'
+
+    def _initialize_pre_login(self):
+        self._HOMEPAGE = f'https://{self._DOMAIN}'
+        self._LOGIN_API = f'{self._HOMEPAGE}/login/'
+        self._LOGOUT_DEVICES_API = f'{self._HOMEPAGE}/logout_devices/?next=/'
+        self._CONTENT_API = f'{self._HOMEPAGE}/api/v2.4/contents/{{content_id}}/'
+        self._HLS_AES_URI = f'{self._HOMEPAGE}/api/v2.5/video_contents/{{content_id}}/key/'
```

Comment:
simpler as class vars
```suggestion
    _HOMEPAGE = f'https://{_DOMAIN}'
    _LOGIN_API = f'{_HOMEPAGE}/login/'
    _LOGOUT_DEVICES_API = f'{_HOMEPAGE}/logout_devices/?next=/'
    _CONTENT_API = f'{_HOMEPAGE}/api/v2.4/contents/{{content_id}}/'
    _HLS_AES_URI = f'{_HOMEPAGE}/api/v2.5/video_contents/{{content_id}}/key/'
```